sdk/src/asset_handlers/jpeg_io.rs

```diff
@@ -69,10 +84,80 @@ fn extract_xmp(seg: &JpegSegment) -> Option<&str> {
 }
 
 // Extract XMP from bytes.
+fn read_be_u32(input: &mut &[u8]) -> Result<u32> {
+    let (int_bytes, rest) = input.split_at(size_of::<u32>());
+    *input = rest;
+    Ok(u32::from_be_bytes(int_bytes.try_into().map_err(|_| {
+        Error::InvalidAsset("Invalid u32 bytes".to_string())
+    })?))
+}
+
+fn extract_xmp_extensions(seg: &JpegSegment) -> Option<XmpExtension> {
+    let (sig, rest) = seg
+        .contents()
+        .split_at_checked(XMP_EXTENSION_SIGNATURE_BUFFER_SIZE)?;
+
+    if sig.starts_with(XMP_EXTENSION_SIGNATURE.as_bytes()) && rest.len() > 40 {
+        // 32 byte GUID, 4 byte length, 4 byte offset
+        let guid = std::str::from_utf8(&rest[..32]).ok()?.to_string();
+        let length = read_be_u32(&mut &rest[32..36]).ok()?;
```

Comment:
Why did you need a new function to read U32 BE
let length = BigEndian::read_u32(&rest[32..36]);